src/servant/tt_003/demon.c

```diff
@@ -511,7 +522,102 @@ void UpdateServantDefault(Entity* self) {
     ServantUpdateAnim(self, &D_us_80171FE8, g_DemonAnimationFrames);
 }
 
-INCLUDE_ASM("servant/tt_003/nonmatchings/demon", func_us_80174FD0);
+void func_us_80174FD0(Entity* self) {
+    if (D_us_801786D4) {
+        self->zPriority = PLAYER.zPriority - 2;
+    }
+    switch (self->step) {
+    case 0:
+        func_us_80172DC4(self);
+        break;
+    case 1:
+        D_us_80178604 = self->ext.demon.target->facingLeft;
+        self->step++;
+        // fallthrough
+    case 2:
+        if (!CheckEntityValid(self->ext.demon.target) &&
+            !(self->ext.demon.target = FindValidTarget(self))) {
+            self->entityId = DEMON_MODE_DEFAULT_UPDATE;
+            self->step = 0;
+        } else {
+            D_us_801785F0 = self->ext.demon.target->posX.val;
+            D_us_801785F0 += D_us_80178604 ? 0x200000 : -0x200000;
```

Comment:
These are FIX I think?

Reply:
Oh yeah. I forgot about FIX. I’ll clean that up here tonight or tomorrow